### PR TITLE
fix: git repo button is unresponsive

### LINKED ISSUE #24070: [Bug]: Open repository button is not functional in git sync modal
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Open repository button is not functional in git sync modal, ideally on clicking this button should take user to git repository 

Screenshot 2023-06-05 at 10 15 36 AM

### Steps To Reproduce

1. Go to git connected app which has conflicts
2. Click on open repository button
3. Observe the click doesnt work

### Public Sample App

_No response_

### Environment

Release

### Issue video log

_No response_

### Version

Release

### PR DESCRIPTION
**Pull request by brayn003**
## Description
Makes go to button clickable

#### PR fixes following issue(s)
Fixes #24070

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [ ] Manual
- [ ] Jest
- [ ] Cypress

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by Parthvi12** [comment]
@brayn003 Should we open repository in new tab instead of the same?

https://github.com/appsmithorg/appsmith/assets/80334441/877587ca-823e-4fee-b397-8f56a19e9c0c

**Comment by brayn003** [comment]
yes

File: app/client/src/pages/Editor/gitSync/components/ConflictInfo.tsx [hunk 1]
```diff
@@ -49,9 +53,9 @@ export default function ConflictInfo({
       <Space size={3} />
       <Row>
         <StyledButton
-          className="t--commit-button"
-          href={browserSupportedRemoteUrl}
+          className="t--git-repo-button"
```
Comment (tanvibhakta) [inline app/client/src/pages/Editor/gitSync/components/ConflictInfo.tsx:56]: can you use a data-testid here?

**Review by tanvibhakta — APPROVED** [review]

**Review by Parthvi12 — APPROVED** [review]

**Pull request merged**